src/tsickle.ts

```diff
@@ -1128,6 +1128,18 @@ class Annotator extends ClosureRewriter {
       this.visit(importClause);
       this.emit(` from '${importPath}';`);
       this.addSourceMapping(decl);
+
+      // Introduce a goog.forwardDeclare for the module, so that if TypeScript does not emit the
+      // module because it's only used in type positions, the JSDoc comments still reference a valid
+      // Closure level symbol.
+      const sym = this.typeChecker.getSymbolAtLocation(decl.moduleSpecifier);
+      if (!sym) {
+        return true;  // modules might not have a symbol if they are unused.
+      }
+      // forwardDeclare all symbols that can be imported from the module.
+      const namedSyms =
+          this.typeChecker.getExportsOfModule(sym).map(sym => ({name: sym.name, sym}));
+      this.forwardDeclare(decl.moduleSpecifier, namedSyms, false);
```

Comment:
Does this mean any module that imports @angular/core ends up with a forwardDeclare line per function in the angular API?

Reply:
No. We emit one `forwardDeclare` (as you can see in the test), but then set all symbols exported by that module to be aliased to `tsickle_forward_declare_X.TheSymbolName`.